### PR TITLE
Use CreateFile2 instead of CreateFile2FromApp

### PR DESCRIPTION
**Pull request by JeremyKuhne**
cc: @joshfree, @davidwrighton, @botaberg, @zamont, @nattress

### CODE REVIEW

**Comment by joshfree** [comment]
Thanks @JeremyKuhne please merge after CI passes

**Comment by JeremyKuhne** [comment]
> Do we want to this in master as well? So that we don't take the FileApiInterop.dll dependency anymore?

Yes, I'll be porting the change after it merges.

File: src/mscorlib/shared/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -557,7 +557,7 @@
   </ItemGroup>
   <ItemGroup Condition="$(TargetsWindows) and ('$(ClrProduct)' == 'core_clr' or '$(EnableWinRT)' == 'true')">
```
Comment (jkotas) [inline src/mscorlib/shared/System.Private.CoreLib.Shared.projitems]: '$(ClrProduct)' == 'core_clr' should not be needed with this change. This should merged into the ` '$(EnableWinRT)' == 'true'` block.

File: src/mscorlib/shared/Interop/Windows/Kernel32/Interop.CreateFile2.cs [hunk 2]
```diff
@@ -10,25 +10,23 @@ internal partial class Interop
 {
     internal partial class FileApiInterop
```
Comment (jkotas) [inline src/mscorlib/shared/Interop/Windows/Kernel32/Interop.CreateFile2.cs]: This should be called Kernel32

Comment (JeremyKuhne) [reply src/mscorlib/shared/Interop/Windows/Kernel32/Interop.CreateFile2.cs]: Thanks

File: src/mscorlib/shared/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (JeremyKuhne) [reply src/mscorlib/shared/System.Private.CoreLib.Shared.projitems]: Fixed

**Comment by JeremyKuhne** [comment]
cc: @danmosemsft

**Comment by joshfree** [comment]
👍

**Pull request merged**